File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        ans = 0 
        j = -1
        lastseen = {}
        for i,c in enumerate(s):
            j = max(j , lastseen.get(c,-1))
            ans = max(ans, i - j)
            lastseen[c] = i
        return ans


        

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int ans = 0 ;
        int j = -1;
        unordered_map<char, int> mpp;
        for(int i=0;i<s.size();i++)
        {
            char c = s[i];
            if(mpp.find(c)!=mpp.end())
            j = max(j,mpp[c]);
            else
            j = max(j,-1);
            ans = max(ans,i-j);
            mpp[c]=i;
        }
        return ans;

        
    }
};